### PR TITLE
DM-12795: Add basic Jenkins CI and stack-os-matrix how-to docs

### PR DESCRIPTION
**Pull request by jonathansick**
Drafts:

- https://developer.lsst.io/v/DM-12795/build-ci/jenkins.html
- https://developer.lsst.io/v/DM-12795/build-ci/jenkins-stack-os-matrix.html
- https://developer.lsst.io/v/DM-12795/processes/workflow.html#testing-with-jenkins

### CODE REVIEW

File: build-ci/jenkins.rst [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+Understanding the weather icons
+-------------------------------
+
+Jenkins represents a job's historical patterns of run successes and failures with weather icons throughout the dashboard interface.
+
+============= ==============================
+Icon          Percent of recent runs passing
+============= ==============================
+Sunny         >80%
```
Comment (SimonKrughoff) [inline build-ci/jenkins.rst]: Is it worth actually putting the icons people will see?  If it's hard to extract the icons, I think the words are understandable.

File: build-ci/jenkins-stack-os-matrix.rst [hunk 2]
```diff
@@ -0,0 +1,60 @@
+##############################################################
+Testing the LSST DM Stack with the Jenkins stack-os-matrix Job
+##############################################################
+
+`stack-os-matrix`_ is the continuous integration (CI) job that DM developers must run to ensure tests pass before merging changes to any LSST EUPS Stack package.
```
Comment (SimonKrughoff) [inline build-ci/jenkins-stack-os-matrix.rst]: before merging any changes to the *master* branch of any LSST EUPS Stack package.

File: build-ci/jenkins-stack-os-matrix.rst [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+   - **Set the list of Git refs** (branch or tag names) to check out from Stack Git repositories.
+
+     For example, if you enter ``tickets/DM-2 tickets/DM-1``, the build system will attempt to check out the ``tickets/DM-2`` branch in Stack packages.
+     If a package does not have a ``tickets/DM-2`` branch, it will attempt to check out the ``tickets/DM-1`` branch.
+     If a package has neither branch, it falls back to checking out the ``master`` branch.
+
+   - **Set the list of EUPS packages** to build.
+     Use the default (``lsst_distrib``) to build and test your changes with a full Stack.
+     To improve build times you can instead specify the name of the package you are actively developing (see also **Skip Demo**, below).
```
Comment (SimonKrughoff) [inline build-ci/jenkins-stack-os-matrix.rst]: Was Skip Demo meant to be a link?  I don't think it has to be, but usually when you say "See something" the something is a link.

File: build-ci/jenkins-stack-os-matrix.rst [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+The progress for each build step is listed on the run detail page.
+The most important build step is ``./buildbot-scripts/jenkins_wrapper.sh`` where the lsstsw rebuild is run.
+Click on that step to see build and test progress for each package.
+
+Viewing and downloading build logs
+==================================
+
+If a build failed, the most effective way to diagnose the problem is to look at the :file:`*.failed` files for individual packages.
+
+You can find these :file:`*.failed` files, along with build logs, under the **Artifacts** tab from the run detail pages.
```
Comment (SimonKrughoff) [inline build-ci/jenkins-stack-os-matrix.rst]: We could link to the Artifacts tab of an example run.  Is that just confusing things?  Or you could add a screen shot, but then I guess we would then need to keep it up to date.

I always have to go hunting for the Artifacts tab.

Comment (timj) [reply build-ci/jenkins-stack-os-matrix.rst]: If a test failed they should be looking at the test report in Jenkins before they go looking for artifacts for the test `.failed` file.

**Comment by jonathansick** [comment]
Thanks, I've made the changes except for the screenshots. It'd be great to have and I might do it as a follow-on ticket but I wanted to merge this in time for the coord meeting.

**Pull request merged**